models/marts/structure/fct_staging_directories.sql

```diff
@@ -23,8 +23,8 @@ staging_models as (
         child,
         child_file_path,
         parent_source_name,
-        left(child_file_path, len(child_file_path) - charindex('/',reverse(child_file_path)) + 1) as child_directory_path,
-        {{ dbt_utils.split_part('child_file_path', "'/'", -1) }} as child_file_name
+        replace(child_file_path, child || '.sql', '') as child_directory_path,
+        regexp_replace(child_file_path,'.*/','') as child_file_name
```

Comment:
genius - thanks for making this way less complicated!

Reply:
To be honest, my first goal was not too make it less complex but just to make it work across all DWs 🤣 

There is an issue currently with `dbt_utils.split_part` where having a negative index only works in Snowflake (the PR to fix it is [here](https://github.com/dbt-labs/dbt-utils/pull/559) but not merged yet)